File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        maps = {
            '}' : '{',
            ']' : '[',
            ')' : '('
        }
        stack = []
        for a in s:
            if a not in maps:
                stack.append(a)
                continue
            if not stack or stack[-1] != maps[a]:
                return False
            stack.pop()
        return not stack

File: Data Structures & Algorithms/minimum-stack/submission-1.swift
class MinStack {
    private var stack = [Int]()

    init() {

    }

    func push(_ val: Int) {
        stack.append(val)
    }

    func pop() {
        stack.popLast()
    }

    func top() -> Int {
        return stack.last!
    }

    func getMin() -> Int {
        var temp = [Int]()
        var mini = stack.last!

        while !stack.isEmpty {
            mini = min(mini, stack.last!)
            temp.append(stack.removeLast())
        }

        while !temp.isEmpty {
            stack.append(temp.removeLast())
        }

        return mini
    }
}
